### DISCUSSION TITLE
Is there a way to install and use two versions of quarto?

### DISCUSSION THREAD

**Discussion by aborruso** [quarto-dev/quarto-cli#5337] [Q&A] [answered]
Hi,
I need to have in my Debian 11, bot 1.3 and 1.4.

I have installed 1.3 via .deb file. Can I install the 1.4 uncompressing quarto-1.4.11-linux-amd64.tar.gz, renaming `quarto` in `quarto14` and simply use it?

Thank you

**Comment by cscheid** [comment, marked as answer]
You can always list the contents of a tarball with `tar t`, like so:

```
$ curl -LO https://github.com/quarto-dev/quarto-cli/releases/download/v1.4.11/quarto-1.4.11-linux-amd64.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 85.8M  100 85.8M    0     0  33.3M      0  0:00:02  0:00:02 --:--:-- 54.3M
$ tar tvzf quarto-1.4.11-linux-amd64.tar.gz
drwxr-xr-x  0 runner docker      0 Apr 27 08:29 quarto-1.4.11/
drwxr-xr-x  0 runner docker      0 Apr 27 08:30 quarto-1.4.11/bin/
-rw-r--r--  0 runner docker 5241679 Apr 27 08:30 quarto-1.4.11/bin/quarto.js
-rwxr-xr-x  0 runner docker    5793 Apr 27 08:29 quarto-1.4.11/bin/quarto
drwxr-xr-x  0 runner docker       0 Apr 27 08:29 quarto-1.4.11/bin/vendor/
-rw-r--r--  0 runner docker    8065 Apr 27 08:29 quarto-1.4.11/bin/vendor/import_map.json
drwxr-xr-x  0 runner docker       0 Apr 27 08:29 quarto-1.4.11/bin/vendor/dev-jspm-io/
-rw-r--r--  0 runner docker     810 Apr 27 08:29 quarto-1.4.11/bin/vendor/dev-jspm-io/npm_pako@1.0.js
...
-rw-r--r--  0 runner docker       377 Apr 27 08:29 quarto-1.4.11/share/extensions/quarto/video/resources/bootstrap/bootstrap-responsive-ratio.css
drwxr-xr-x  0 runner docker         0 Apr 27 08:29 quarto-1.4.11/share/extensions/quarto/kbd/
-rw-r--r--  0 runner docker        92 Apr 27 08:29 quarto-1.4.11/share/extensions/quarto/kbd/_extension.yml
-rw-r--r--  0 runner docker      2915 Apr 27 08:29 quarto-1.4.11/share/extensions/quarto/kbd/kbd.lua
drwxr-xr-x  0 runner docker         0 Apr 27 08:29 quarto-1.4.11/share/extensions/quarto/kbd/resources/
-rw-r--r--  0 runner docker        12 Apr 27 08:29 quarto-1.4.11/share/extensions/quarto/kbd/resources/kbd.css
-rw-r--r--  0 runner docker      1000 Apr 27 08:29 quarto-1.4.11/share/extensions/quarto/kbd/resources/kbd.js
```

As you can see, the contents are all inside `quarto-1.4.11`. So if you extract that tarball, you'll find

```
$ ./quarto-1.4.11/bin/quarto --version
1.4.11
```

So just download the other tarballs and run them from their respective directories.

**Reply by aborruso** [reply]
> You can always list the contents of a tarball with `tar t`, like so:

I know I have quarto inside it, thank you 🙃

I wasn't sure if simply extracting the compressed file, making the file executable, and adding the folder to the PATH was enough.
I understand that it is. 

Thank you

**Reply by aborruso** [reply]
> So just download the other tarballs and run them from their respective directories.

I have inserted it in the path. And then I have set `chmod +x` for

- `/home/aborruso/bin/quarto/bin/tools/pandoc`
- `/home/aborruso/bin/quarto/bin/tools/dart-sass/sass`
- `/home/aborruso/bin/quarto/bin/tools/deno-x86_64-unknown-linux-gnu/deno`

@cscheid  Probably nothing else is required. Am I wrong?

**Reply by cscheid** [reply]
You shouldn't need to make any files executable, tar should preserve the flags. And instead of adding the folder to PATH, I would find a folder in your path like `~/local/bin`, and create a symbolic link from there to the (eg) `./quarto-1.4.11/bin/quarto` script.

**Comment by cderv** [comment]
I guess the tools https://github.com/dpastoor/qvm is also a good way to install and switch between several quarto versions

**Reply by aborruso** [reply]
> I guess the tools [dpastoor/qvm](https://github.com/dpastoor/qvm) is also a good way to install and switch between several quarto versions

thank you very much
